[8.0] PXB-2724 : Make --move-back work with RocksDB engine

jakub-nowakowski-percona opened the pull request:
https://perconadev.atlassian.net/browse/PXB-2724

Problem & Fix:
--------------
--move-back option hasn't been used when restoring RocksDB files, which always resulted in a copy. Now, the missing check and related move code have been added.

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/signed)](https://cla-assistant.percona.com:443/percona/percona-xtrabackup?pullRequest=1694) <br/>All committers have signed the CLA.

jakub-nowakowski-percona committed:
PXB-2724 : Make --move-back work with RocksDB engine

https://perconadev.atlassian.net/browse/PXB-2724

Problem & Fix:
--------------
--move-back option hasn't been used when restoring RocksDB files, which always
resulted in a copy. Now, the missing check and related move code have been
added.

jakub-nowakowski-percona commented:
Tests: https://pxb.cd.percona.com/view/PXB%208.0/job/percona-xtrabackup-8.0-multijob/370/

satya-bodapati reviewed (approved):
LGTM!
